File: Asklepios/Asklepios.Core/Entities/Departments/Department.cs
using Asklepios.Core.Entities.Patients;
using Asklepios.Core.Entities.Users;

namespace Asklepios.Core.Entities.Departments;

public class Department
{
    public Guid DepartmentId { get; set; }
    public string DepartmentName { get; set; }
    public int NumberOfBeds { get; set; }
    public int ActualNumberOfPatients { get; set; }
    public IEnumerable<Room> Rooms { get; set; }
    public IEnumerable<Patient> Patients { get; set; }
    
    public Nurse Nurse { get; set; }
    public MedicalStaff MedicalStaff { get; set; }

    public Department(Guid departmentId, string departmentName, int numberOfBeds, int actualNumberOfPatients)
    {
        DepartmentId = departmentId;
        DepartmentName = departmentName;
        NumberOfBeds = numberOfBeds;
        ActualNumberOfPatients = actualNumberOfPatients;
    }

    public Department()
    {
    }
}

File: Asklepios.Web/src/models/Departments/department.ts
import {RoomDto} from "@/models/Departments/room"
import {PatientDto} from "@/models/Patients/patient"

export interface Department {
  departmentId: string
  departmentName: string
  numberOfBeds: number
  actualNumberOfPatients: number
}

export class InputCreateDepartment {
  departmentName: string = ''
  numberOfBeds: number = 0
  actualNumberOfPatients: number = 0
}

export interface DepartmentDetailsDto extends DepartmentDto {
  rooms: RoomDto[];
  patients: PatientDto[];
}

export interface DepartmentDto {
  departmentId: string;
  departmentName: string;
  numberOfBeds: number;
  actualNumberOfPatients: number;
}


export interface DepartmentListDto {
  departmentId: string;
  departmentName: string;
  numberOfBeds: number;
  actualNumberOfPatients: number;
}
